switches/http/http.cpp

```diff
@@ -252,84 +271,221 @@ namespace Aseba
         }
     }
     
-    void HttpInterface::sendMessage(const Message& message)
-	{
-		message.serialize(asebaStream);
-		asebaStream->flush();
-	}
+    bool HttpInterface::run1s()
+    {
+        int timeout(1000);
+        UnifiedTime startTime;
+        while (timeout > 0)
+        {
+            // special handling for HTTP streams
+            sendAvailableResponses();
+            if (verbose && streamsToShutdown.size() > 0)
+            {
+                cerr << "HttpInterface::run "<< streamsToShutdown.size() <<" streams to shut down";
+                for (StreamSet::iterator si = streamsToShutdown.begin(); si != streamsToShutdown.end(); si++)
+                    cerr << " " << *si;
+                cerr << endl;
+            }
+            if (!streamsToShutdown.empty())
+            {
+                StreamSet::iterator i = streamsToShutdown.begin();
+                Dashel::Stream* stream_to_shutdown = *i;
+                streamsToShutdown.erase(*i); // invalidates iterator
+                try
+                {
+                    if (verbose)
+                        cerr << stream_to_shutdown << " shutting down stream" << endl;
+                    shutdownStream(stream_to_shutdown);
+                }
+                catch(Dashel::DashelException& e)
+                { }
+            }
+
+            // standard Aseba run loop
+            if (!step(timeout))
```

Comment:
Using `timeout` seems to throttle communications, I get better results with a low value like `50`.